flexdll.c

```diff
@@ -403,6 +406,20 @@ void *flexdll_dlopen(const char *file, int mode) {
   return unit;
 }
 
+void *flexdll_dlopen(const char *file, int mode)
+{
+  int exec = mode & FLEXDLL_RTLD_NOEXEC ? 0 : 1;
+  if (!file) return &main_unit;
+  return flexdll_dlopen_aux(ll_dlopen(file, exec), mode);
```

Comment:
I think this is wrong: ll_dlopen would now be called before setting FLEXDLL_RELOCATE, which can be read by the DLL's entry point.